lib/index.js

```diff
@@ -181,6 +181,135 @@ function defaultContext() {
 };
 
 /**
+ */
+
+function OutMessage(buf, flags, cb) {
+  if (!Buffer.isBuffer(buf)) {
+    buf = new Buffer(String(buf), 'utf8');
+  }
+
+  this.buf = buf;
+  this.flags = flags | 0;
+  this.cb = cb;
+  this.next = null;
+}
+
+/**
+ * Message chain, a linked list of message batches.
+ * A batch is 1 or more messages that need to be sent as one unit (SNDMORE until, not including, the last message)
+ */
```

Comment:
I need to update and augment these comments. Can also externalize a lot of this code in separate files.